### PR TITLE
Setup event response with reactions

### PR DESCRIPTION
**Pull request by jgayfer**
This PR adds functionality for users to indicate a response to an event (accepted, declined, or alternate). When a valid reaction is added, the database is updated with the updated response.

Aside from the reactions themselves, the changes are only in the back end at this point; the event is not yet updated to indicate the user's response.

#### To verify this change:

1. Create an event
2. Click on the reactions that are added to the event (multiple times)
3. Observe that the database is updated with new attendance value (shown in console)

### CODE REVIEW

File: event_bot/events/on_raw_reaction_add.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+    def _find_existing_response(self, event, user_id):
+        with self.bot.transaction.new() as session:
+            return session.query(Response).filter_by(event_id=event.id, user_id=user_id).first()
+
+
+    def _get_response_status(self, emoji):
+        if emoji.name == ACCEPTED:
+            return 'accepted'
```
Comment (seriousammy) [inline event_bot/events/on_raw_reaction_add.py]: In the past I've learned that using strings this way is a bad practice? I'm not 100% sure. But for stuff like this...especially in classes you should do at least something liek this?
```
...
from event_bot.constants import * as Emoji
...
class OnRawReactionAdd:
    DECLINED = 'declined'
    ACCEPTED = 'accepted'
    ALTERNATE = 'alternate'
    ...
    def _get_response_status(self, emoji):
          if emoji.name == Emoji.ACCEPTED:
                return ACCEPTED
    ....
```
excuse the horrible spacing in this example. 
This way if we ever need to make a comparison we can do `OnRawReactionAdd.DECLINED` rrather than a comparison to `"declined"`
I might also be missing out on the bigger scope, let me know your thoughts.

**Comment by seriousammy** [comment]
I liked that you pulled out and made a new file called `constants` was going to meniton it on a few PRs ago when u had that skull one, but ... no need to make a new file for just one const lol. so this is great, that you made one as you started to accumlate more emoji/constants.

One thing to consider is to bring in a choices list in the class? I'm not great at explaining but something like this, (this sort of goes along with my comment in the PR): 

```
...
from event_bot.constants import * as Emoji
...
class OnRawReactionAdd:
    DECLINED = 'declined'
    ACCEPTED = 'accepted'
    ALTERNATE = 'alternate'
    REACTION_CHOICES = [
        (DECLINED, Emoji.DECLINED),
        (ACCEPTED, Emoji.ACCEPTED),
        (ALTERNATE, Emoji.ALTERNATE)
    ]
    ...
    def _get_response_status(self, emoji):
          ...
    ....
```
Thoughts?

File: event_bot/events/on_raw_reaction_add.py (hunk 1 above)
Comment (jgayfer) [reply event_bot/events/on_raw_reaction_add.py]: I like this idea. You're definitely right, naked strings are bad practice.

**Comment by jgayfer** [comment]
A dict makes perefect sense actually.

```
emoji_statuses = {Emoji.ACCEPTED: ACCEPTED,
Emoji.DECLINED: DECLINED,
Emoji.ALTERNATE: ALTERNATE}

# Then we can just have one line to get the status
response_status = emoji_statuses[emoji.name]

```

**Comment by jgayfer** [comment]
@seriousammy Did a pretty significant rework here. Ready for a re-review!

**Pull request merged**